### ISSUE TITLE
`function_error` is nil even when there is an error raised in lambda

### ISSUE THREAD

**Issue by invacuo** [awslabs/aws-sam-cli#1148]
### Description

I initially opened this issue in aws-sdk-ruby repo but was told to post it here.  Here is a link to the original issue post https://github.com/aws/aws-sdk-ruby/issues/2032

The `function_error` or `log_result` is not populated in the response when the client is pointing to a locally running lambda started using `sam local start-lambda`

Briefly describe the bug you are facing.

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

`aws-sdk-lambda (1.21.0)`

### Version of Ruby, OS environment
`ruby-2.5.1` MacOS Mojave

### Code snippets / steps to reproduce
I have a following simple lambda
```ruby
require 'json'

def lambda_handler(event:, context:)
  raise ArgumentError.new('Missing key error: foo') unless event.key?('foo')

  {
    message: "Hello #{event['foo']}!"
  }.to_json
rescue ArgumentError => e
  # we will log a honeybadger error here
  puts 'event value'
  puts event
  raise e
end
```

I am using rspec to test this.

This works fine
```ruby
        expect do
          lambda_handler(event: {}, context: '')
        end.to raise_error(ArgumentError)
```

The following invocation does not populate `function_error` or `log_result` in the response though

```ruby

def lambda_client
    @lambda_client ||= Aws::Lambda::Client.new(
      endpoint: 'http://127.0.0.1:3001/'
    )
end


.
.
.
.
.
.

# In my actual spec
          event = { 'boo' => 'far' }

          resp = lambda_client.invoke({
            function_name: "HelloWorldFunction",
            invocation_type: "RequestResponse",
            payload:  event.to_json
          })
```

Here is the output in the console after. Notice `resp.payload.string` is blank as well which is correct as I am raising an error as the first line.

```shell
[1] pry(#<RSpec::ExampleGroups::LambdaTest::LambdaHandler::CallingItViaLambdaInvoke::InvalidEventParamsPassed>)> resp
=> #<struct Aws::Lambda::Types::InvocationResponse status_code=200, function_error=nil, log_result=nil, payload=#<StringIO:0x00007fb28a0a74d8>, executed_version=nil>
[2] pry(#<RSpec::ExampleGroups::LambdaTest::LambdaHandler::CallingItViaLambdaInvoke::InvalidEventParamsPassed>)> resp.payload.string
=> ""
```

This positive test case passes

```ruby
          event = { 'foo' => 'bar' }

          resp = lambda_client.invoke({
            function_name: "HelloWorldFunction",
            invocation_type: "RequestResponse",
            log_type: "None",
            payload:  event.to_json
          })
          response_payload = JSON.parse(resp.payload.string)

          expect(resp.status_code).to eq 200
          expect(response_payload).to eq ({
            message: "Hello #{event['foo']}!"
          }.to_json)
```

I do see the request being processed where I have `sam local start-lambda` running. See the output below.

```bash
Fetching lambci/lambda:ruby2.5 Docker container image......
2019-04-23 16:51:44 Mounting /Users/chaitanya.kale/workspace/sam-app/hello_world as /var/task:ro,delegated inside runtime container
START RequestId: 52fdfc07-2182-154f-163f-5f0f9a621d72 Version: $LATEST
event value
{"boo"=>"far"}
Error raised from handler method
{
  "errorMessage": "Missing key error: foo",
  "errorType": "Function<ArgumentError>",
  "stackTrace": [
    "/var/task/app.rb:31:in `lambda_handler'"
  ]
}
END RequestId: 52fdfc07-2182-154f-163f-5f0f9a621d72
REPORT RequestId: 52fdfc07-2182-154f-163f-5f0f9a621d72	Duration: 2.96 ms	Billed Duration: 100 ms	Memory Size: 128 MB	Max Memory Used: 18 MB
{
  "errorType": "Function\u003cArgumentError\u003e",
  "errorMessage": "Missing key error: foo",
  "stackTrace": [
    "/var/task/app.rb:31:in `lambda_handler'"
  ]
}
```

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: MacOS Mojave
2. `sam --version`: `SAM CLI, version 0.15.0`

**Comment by awood45** [comment]
Ok, I am able to reproduce this.

Running on AWS Lambda:

```
pry(main)> resp = lambda.invoke(function_name: function_name, invocation_type: "RequestResponse", payload: "{}")
=> #<struct Aws::Lambda::Types::InvocationResponse status_code=200, function_error="Unhandled", log_result=nil, payload=#<StringIO:0x00007f9bed1cdfc8>, executed_version="$LATEST">
```

Running versus Lambda Local:

```
pry(main)> resp = lambda.invoke(function_name: "HelloWorldFunction", invocation_type: "RequestResponse", payload: "{}")
=> #<struct Aws::Lambda::Types::InvocationResponse status_code=200, function_error=nil, log_result=nil, payload=#<StringIO:0x00007f9bed24cb20>, executed_version=nil>
```

**Comment by invacuo** [comment]
Thanks @awood45 for looking into it!

**Comment by jfuss** [comment]
This looks to be addressed by #1642, which was released in [v0.38.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.38.0).

Running the above code, I am now seeing this response:
```
➜  ~ aws lambda invoke --function-name HelloWorldFunction --endpoint-url http://127.0.0.1:3001 --no-verify-ssl --region us-west-2 output.lambda --no-sign-request
{
    "StatusCode": 200,
    "FunctionError": "Unhandled"
}
```

Which matches the expectation above from Lambda.

Closing as this has been released but if this is incorrect, please feel free to reopen or create a new issue.
